[2025-04-14T00:00:01.348+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-04-14T00:00:01.363+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: weather_forecast_ingestion_dag.ingest scheduled__2025-04-13T23:00:00+00:00 [queued]>
[2025-04-14T00:00:01.368+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: weather_forecast_ingestion_dag.ingest scheduled__2025-04-13T23:00:00+00:00 [queued]>
[2025-04-14T00:00:01.369+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 2
[2025-04-14T00:00:01.376+0000] {taskinstance.py:2890} INFO - Executing <Task(_PythonDecoratedOperator): ingest> on 2025-04-13 23:00:00+00:00
[2025-04-14T00:00:01.383+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=615) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-04-14T00:00:01.388+0000] {standard_task_runner.py:72} INFO - Started process 617 to run task
[2025-04-14T00:00:01.387+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'weather_forecast_ingestion_dag', 'ingest', 'scheduled__2025-04-13T23:00:00+00:00', '--job-id', '956', '--raw', '--subdir', 'DAGS_FOLDER/weather_forecast_ingestion_dag.py', '--cfg-path', '/tmp/tmp4pnj99dl']
[2025-04-14T00:00:01.390+0000] {standard_task_runner.py:105} INFO - Job 956: Subtask ingest
[2025-04-14T00:00:01.439+0000] {task_command.py:467} INFO - Running <TaskInstance: weather_forecast_ingestion_dag.ingest scheduled__2025-04-13T23:00:00+00:00 [running]> on host caedfbfc7792
[2025-04-14T00:00:01.520+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='beni' AIRFLOW_CTX_DAG_ID='weather_forecast_ingestion_dag' AIRFLOW_CTX_TASK_ID='ingest' AIRFLOW_CTX_EXECUTION_DATE='2025-04-13T23:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-04-13T23:00:00+00:00'
[2025-04-14T00:00:01.521+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-04-14T00:00:01.523+0000] {weather_forecast_ingestion_dag.py:37} INFO - 🌤️ Fetching forecast from 2025-04-14 to 2025-04-22
[2025-04-14T00:00:02.233+0000] {weather_forecast_ingestion_dag.py:53} INFO - 📦 Filtered 120 future forecast entries
[2025-04-14T00:00:02.234+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-14T06:00:00+00:00
[2025-04-14T00:00:02.726+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-14T0600.json
[2025-04-14T00:00:02.727+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-14T07:00:00+00:00
[2025-04-14T00:00:03.123+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-14T0700.json
[2025-04-14T00:00:03.124+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-14T08:00:00+00:00
[2025-04-14T00:00:03.533+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-14T0800.json
[2025-04-14T00:00:03.534+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-14T09:00:00+00:00
[2025-04-14T00:00:03.919+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-14T0900.json
[2025-04-14T00:00:03.920+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-14T10:00:00+00:00
[2025-04-14T00:00:05.211+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-14T1000.json
[2025-04-14T00:00:05.211+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-14T11:00:00+00:00
[2025-04-14T00:00:05.566+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-14T1100.json
[2025-04-14T00:00:05.568+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-14T12:00:00+00:00
[2025-04-14T00:00:05.978+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-14T1200.json
[2025-04-14T00:00:05.978+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-14T13:00:00+00:00
[2025-04-14T00:00:06.647+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-14T1300.json
[2025-04-14T00:00:06.648+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-14T14:00:00+00:00
[2025-04-14T00:00:07.074+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-14T1400.json
[2025-04-14T00:00:07.074+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-14T15:00:00+00:00
[2025-04-14T00:00:07.443+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-14T1500.json
[2025-04-14T00:00:07.444+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-14T16:00:00+00:00
[2025-04-14T00:00:07.829+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-14T1600.json
[2025-04-14T00:00:07.830+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-14T17:00:00+00:00
[2025-04-14T00:00:08.218+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-14T1700.json
[2025-04-14T00:00:08.219+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-14T18:00:00+00:00
[2025-04-14T00:00:08.618+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-14T1800.json
[2025-04-14T00:00:08.619+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-14T19:00:00+00:00
[2025-04-14T00:00:08.973+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-14T1900.json
[2025-04-14T00:00:08.973+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-14T20:00:00+00:00
[2025-04-14T00:00:09.350+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-14T2000.json
[2025-04-14T00:00:09.351+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-15T06:00:00+00:00
[2025-04-14T00:00:09.732+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-15T0600.json
[2025-04-14T00:00:09.733+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-15T07:00:00+00:00
[2025-04-14T00:00:10.090+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-15T0700.json
[2025-04-14T00:00:10.091+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-15T08:00:00+00:00
[2025-04-14T00:00:10.475+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-15T0800.json
[2025-04-14T00:00:10.476+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-15T09:00:00+00:00
[2025-04-14T00:00:10.899+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-15T0900.json
[2025-04-14T00:00:10.901+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-15T10:00:00+00:00
[2025-04-14T00:00:11.311+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-15T1000.json
[2025-04-14T00:00:11.312+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-15T11:00:00+00:00
[2025-04-14T00:00:11.701+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-15T1100.json
[2025-04-14T00:00:11.703+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-15T12:00:00+00:00
[2025-04-14T00:00:12.086+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-15T1200.json
[2025-04-14T00:00:12.087+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-15T13:00:00+00:00
[2025-04-14T00:00:12.456+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-15T1300.json
[2025-04-14T00:00:12.457+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-15T14:00:00+00:00
[2025-04-14T00:00:12.814+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-15T1400.json
[2025-04-14T00:00:12.815+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-15T15:00:00+00:00
[2025-04-14T00:00:13.210+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-15T1500.json
[2025-04-14T00:00:13.212+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-15T16:00:00+00:00
[2025-04-14T00:00:13.608+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-15T1600.json
[2025-04-14T00:00:13.609+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-15T17:00:00+00:00
[2025-04-14T00:00:13.975+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-15T1700.json
[2025-04-14T00:00:13.977+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-15T18:00:00+00:00
[2025-04-14T00:17:51.279+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-15T1800.json
[2025-04-14T00:17:51.280+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-15T19:00:00+00:00
[2025-04-14T00:17:51.681+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-15T1900.json
[2025-04-14T00:17:51.681+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-15T20:00:00+00:00
[2025-04-14T00:17:52.055+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-15T2000.json
[2025-04-14T00:17:52.055+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-16T06:00:00+00:00
[2025-04-14T00:17:52.417+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-16T0600.json
[2025-04-14T00:17:52.418+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-16T07:00:00+00:00
[2025-04-14T00:17:52.791+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-16T0700.json
[2025-04-14T00:17:52.792+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-16T08:00:00+00:00
[2025-04-14T00:17:53.142+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-16T0800.json
[2025-04-14T00:17:53.142+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-16T09:00:00+00:00
[2025-04-14T00:17:53.431+0000] {job.py:229} INFO - Heartbeat recovered after 1061.98 seconds
[2025-04-14T00:17:53.471+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-16T0900.json
[2025-04-14T00:17:53.472+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-16T10:00:00+00:00
[2025-04-14T00:17:53.848+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-16T1000.json
[2025-04-14T00:17:53.849+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-16T11:00:00+00:00
[2025-04-14T00:17:54.227+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-16T1100.json
[2025-04-14T00:17:54.229+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-16T12:00:00+00:00
[2025-04-14T00:17:54.605+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-16T1200.json
[2025-04-14T00:17:54.605+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-16T13:00:00+00:00
[2025-04-14T00:17:54.994+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-16T1300.json
[2025-04-14T00:17:54.995+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-16T14:00:00+00:00
[2025-04-14T00:17:55.396+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-16T1400.json
[2025-04-14T00:17:55.398+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-16T15:00:00+00:00
[2025-04-14T00:17:55.776+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-16T1500.json
[2025-04-14T00:17:55.777+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-16T16:00:00+00:00
[2025-04-14T00:17:56.134+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-16T1600.json
[2025-04-14T00:17:56.135+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-16T17:00:00+00:00
[2025-04-14T00:17:56.533+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-16T1700.json
[2025-04-14T00:17:56.533+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-16T18:00:00+00:00
[2025-04-14T00:17:56.932+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-16T1800.json
[2025-04-14T00:17:56.934+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-16T19:00:00+00:00
[2025-04-14T00:17:57.305+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-16T1900.json
[2025-04-14T00:17:57.306+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-16T20:00:00+00:00
[2025-04-14T00:17:57.671+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-16T2000.json
[2025-04-14T00:17:57.671+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-17T06:00:00+00:00
[2025-04-14T00:17:58.038+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-17T0600.json
[2025-04-14T00:17:58.039+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-17T07:00:00+00:00
[2025-04-14T00:17:58.403+0000] {logging_mixin.py:190} INFO - 🌤️ Saved forecast: gs://gooutside-raw/weather_forecast/bamberg/2025-04-17T0700.json
[2025-04-14T00:17:58.404+0000] {weather_forecast_ingestion_dag.py:56} INFO - 💾 Saving forecast for 2025-04-17T08:00:00+00:00
[2025-04-14T00:17:58.467+0000] {local_task_job_runner.py:346} WARNING - State of this instance has been externally set to up_for_retry. Terminating instance.
[2025-04-14T00:17:58.469+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-04-14T00:17:58.470+0000] {process_utils.py:132} INFO - Sending 15 to group 617. PIDs of all processes in the group: [617]
[2025-04-14T00:17:58.470+0000] {process_utils.py:87} INFO - Sending the signal 15 to group 617
[2025-04-14T00:17:58.475+0000] {taskinstance.py:3094} ERROR - Received SIGTERM. Terminating subprocesses.
[2025-04-14T00:17:58.484+0000] {taskinstance.py:3095} ERROR - Stacktrace: 
  File "/home/***/.local/bin/***", line 8, in <module>
    sys.exit(main())
  File "/home/***/.local/lib/python3.12/site-packages/***/__main__.py", line 62, in main
    args.func(args)
  File "/home/***/.local/lib/python3.12/site-packages/***/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/***/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/***/providers/celery/cli/celery_command.py", line 64, in wrapper
    providers_configuration_loaded(func)(*args, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/***/utils/providers_configuration_loader.py", line 55, in wrapped_function
    return func(*args, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/***/providers/celery/cli/celery_command.py", line 237, in worker
    _run_command_with_daemon_option(
  File "/home/***/.local/lib/python3.12/site-packages/***/providers/celery/cli/celery_command.py", line 50, in _run_command_with_daemon_option
    run_command_with_daemon_option(*args, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/***/cli/commands/daemon_utils.py", line 86, in run_command_with_daemon_option
    callback()
  File "/home/***/.local/lib/python3.12/site-packages/***/providers/celery/cli/celery_command.py", line 230, in run_celery_worker
    celery_app.worker_main(options)
  File "/home/***/.local/lib/python3.12/site-packages/celery/app/base.py", line 391, in worker_main
    self.start(argv=argv)
  File "/home/***/.local/lib/python3.12/site-packages/celery/app/base.py", line 371, in start
    celery.main(args=argv, standalone_mode=False)
  File "/home/***/.local/lib/python3.12/site-packages/click/core.py", line 1082, in main
    rv = self.invoke(ctx)
  File "/home/***/.local/lib/python3.12/site-packages/click/core.py", line 1697, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/***/.local/lib/python3.12/site-packages/click/core.py", line 1443, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/***/.local/lib/python3.12/site-packages/click/core.py", line 788, in invoke
    return __callback(*args, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/celery/bin/base.py", line 135, in caller
    return f(ctx, *args, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/celery/bin/worker.py", line 356, in worker
    worker.start()
  File "/home/***/.local/lib/python3.12/site-packages/celery/worker/worker.py", line 202, in start
    self.blueprint.start(self)
  File "/home/***/.local/lib/python3.12/site-packages/celery/bootsteps.py", line 116, in start
    step.start(parent)
  File "/home/***/.local/lib/python3.12/site-packages/celery/bootsteps.py", line 365, in start
    return self.obj.start()
  File "/home/***/.local/lib/python3.12/site-packages/celery/concurrency/base.py", line 130, in start
    self.on_start()
  File "/home/***/.local/lib/python3.12/site-packages/celery/concurrency/prefork.py", line 109, in on_start
    P = self._pool = Pool(processes=self.limit,
  File "/home/***/.local/lib/python3.12/site-packages/celery/concurrency/asynpool.py", line 464, in __init__
    super().__init__(processes, *args, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/billiard/pool.py", line 1046, in __init__
    self._create_worker_process(i)
  File "/home/***/.local/lib/python3.12/site-packages/celery/concurrency/asynpool.py", line 482, in _create_worker_process
    return super()._create_worker_process(i)
  File "/home/***/.local/lib/python3.12/site-packages/billiard/pool.py", line 1158, in _create_worker_process
    w.start()
  File "/home/***/.local/lib/python3.12/site-packages/billiard/process.py", line 120, in start
    self._popen = self._Popen(self)
  File "/home/***/.local/lib/python3.12/site-packages/billiard/context.py", line 331, in _Popen
    return Popen(process_obj)
  File "/home/***/.local/lib/python3.12/site-packages/billiard/popen_fork.py", line 22, in __init__
    self._launch(process_obj)
  File "/home/***/.local/lib/python3.12/site-packages/billiard/popen_fork.py", line 77, in _launch
    code = process_obj._bootstrap()
  File "/home/***/.local/lib/python3.12/site-packages/billiard/process.py", line 323, in _bootstrap
    self.run()
  File "/home/***/.local/lib/python3.12/site-packages/billiard/process.py", line 110, in run
    self._target(*self._args, **self._kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/billiard/pool.py", line 292, in __call__
    sys.exit(self.workloop(pid=pid))
  File "/home/***/.local/lib/python3.12/site-packages/billiard/pool.py", line 362, in workloop
    result = (True, prepare_result(fun(*args, **kwargs)))
  File "/home/***/.local/lib/python3.12/site-packages/celery/app/trace.py", line 651, in fast_trace_task
    R, I, T, Rstr = tasks[task].__trace__(
  File "/home/***/.local/lib/python3.12/site-packages/celery/app/trace.py", line 453, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/celery/app/trace.py", line 736, in __protected_call__
    return self.run(*args, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/***/providers/celery/executors/celery_executor_utils.py", line 139, in execute_command
    _execute_in_fork(command_to_exec, celery_task_id)
  File "/home/***/.local/lib/python3.12/site-packages/***/providers/celery/executors/celery_executor_utils.py", line 171, in _execute_in_fork
    args.func(args)
  File "/home/***/.local/lib/python3.12/site-packages/***/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/***/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/***/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/***/.local/lib/python3.12/site-packages/***/cli/commands/task_command.py", line 254, in _run_task_by_selected_method
    return _run_task_by_local_task_job(args, ti)
  File "/home/***/.local/lib/python3.12/site-packages/***/cli/commands/task_command.py", line 322, in _run_task_by_local_task_job
    ret = run_job(job=job_runner.job, execute_callable=job_runner._execute)
  File "/home/***/.local/lib/python3.12/site-packages/***/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/***/jobs/job.py", line 421, in run_job
    return execute_job(job, execute_callable=execute_callable)
  File "/home/***/.local/lib/python3.12/site-packages/***/jobs/job.py", line 450, in execute_job
    ret = execute_callable()
  File "/home/***/.local/lib/python3.12/site-packages/***/jobs/local_task_job_runner.py", line 171, in _execute
    self.task_runner.start()
  File "/home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py", line 55, in start
    self.process = self._start_by_fork()
  File "/home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/home/***/.local/lib/python3.12/site-packages/***/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/***/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/***/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/***/.local/lib/python3.12/site-packages/***/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/home/***/.local/lib/python3.12/site-packages/***/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/home/***/.local/lib/python3.12/site-packages/***/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/***/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
  File "/home/***/.local/lib/python3.12/site-packages/***/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/***/.local/lib/python3.12/site-packages/***/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/home/***/.local/lib/python3.12/site-packages/***/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/home/***/.local/lib/python3.12/site-packages/***/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/***/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/***/.local/lib/python3.12/site-packages/***/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/***/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/***/decorators/base.py", line 266, in execute
    return_value = super().execute(context)
  File "/home/***/.local/lib/python3.12/site-packages/***/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/***/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/***/.local/lib/python3.12/site-packages/***/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/***/.local/lib/python3.12/site-packages/***/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/***/dags/weather_forecast_ingestion_dag.py", line 57, in ingest
    save_weather_forecast_to_gcs(entry)
  File "/opt/***/app/ingestion/weather_forecast.py", line 74, in save_weather_forecast_to_gcs
    client = storage.Client()
  File "/home/***/.local/lib/python3.12/site-packages/google/cloud/storage/client.py", line 228, in __init__
    super(Client, self).__init__(
  File "/home/***/.local/lib/python3.12/site-packages/google/cloud/client/__init__.py", line 320, in __init__
    _ClientProjectMixin.__init__(self, project=project, credentials=credentials)
  File "/home/***/.local/lib/python3.12/site-packages/google/cloud/client/__init__.py", line 268, in __init__
    project = self._determine_default(project)
  File "/home/***/.local/lib/python3.12/site-packages/google/cloud/client/__init__.py", line 287, in _determine_default
    return _determine_default_project(project)
  File "/home/***/.local/lib/python3.12/site-packages/google/cloud/_helpers/__init__.py", line 152, in _determine_default_project
    _, project = google.auth.default()
  File "/home/***/.local/lib/python3.12/site-packages/google/auth/_default.py", line 685, in default
    credentials, project_id = checker()
  File "/home/***/.local/lib/python3.12/site-packages/google/auth/_default.py", line 678, in <lambda>
    lambda: _get_explicit_environ_credentials(quota_project_id=quota_project_id),
  File "/home/***/.local/lib/python3.12/site-packages/google/auth/_default.py", line 293, in _get_explicit_environ_credentials
    credentials, project_id = load_credentials_from_file(
  File "/home/***/.local/lib/python3.12/site-packages/google/auth/_default.py", line 137, in load_credentials_from_file
    return _load_credentials_from_info(
  File "/home/***/.local/lib/python3.12/site-packages/google/auth/_default.py", line 210, in _load_credentials_from_info
    credentials, project_id = _get_service_account_credentials(
  File "/home/***/.local/lib/python3.12/site-packages/google/auth/_default.py", line 473, in _get_service_account_credentials
    credentials = service_account.Credentials.from_service_account_info(
  File "/home/***/.local/lib/python3.12/site-packages/google/oauth2/service_account.py", line 243, in from_service_account_info
    signer = _service_account_info.from_dict(
  File "/home/***/.local/lib/python3.12/site-packages/google/auth/_service_account_info.py", line 57, in from_dict
    signer = crypt.RSASigner.from_service_account_info(data)
  File "/home/***/.local/lib/python3.12/site-packages/google/auth/crypt/base.py", line 109, in from_service_account_info
    return cls.from_string(
  File "/home/***/.local/lib/python3.12/site-packages/google/auth/crypt/_cryptography_rsa.py", line 133, in from_string
    private_key = serialization.load_pem_private_key(
  File "/home/***/.local/lib/python3.12/site-packages/***/models/taskinstance.py", line 3095, in signal_handler
    self.log.error("Stacktrace: \n%s", "".join(traceback.format_stack()))

[2025-04-14T00:17:58.492+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/decorators/base.py", line 266, in execute
    return_value = super().execute(context)
                   ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/weather_forecast_ingestion_dag.py", line 57, in ingest
    save_weather_forecast_to_gcs(entry)
  File "/opt/airflow/app/ingestion/weather_forecast.py", line 74, in save_weather_forecast_to_gcs
    client = storage.Client()
             ^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/google/cloud/storage/client.py", line 228, in __init__
    super(Client, self).__init__(
  File "/home/airflow/.local/lib/python3.12/site-packages/google/cloud/client/__init__.py", line 320, in __init__
    _ClientProjectMixin.__init__(self, project=project, credentials=credentials)
  File "/home/airflow/.local/lib/python3.12/site-packages/google/cloud/client/__init__.py", line 268, in __init__
    project = self._determine_default(project)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/google/cloud/client/__init__.py", line 287, in _determine_default
    return _determine_default_project(project)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/google/cloud/_helpers/__init__.py", line 152, in _determine_default_project
    _, project = google.auth.default()
                 ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/google/auth/_default.py", line 685, in default
    credentials, project_id = checker()
                              ^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/google/auth/_default.py", line 678, in <lambda>
    lambda: _get_explicit_environ_credentials(quota_project_id=quota_project_id),
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/google/auth/_default.py", line 293, in _get_explicit_environ_credentials
    credentials, project_id = load_credentials_from_file(
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/google/auth/_default.py", line 137, in load_credentials_from_file
    return _load_credentials_from_info(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/google/auth/_default.py", line 210, in _load_credentials_from_info
    credentials, project_id = _get_service_account_credentials(
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/google/auth/_default.py", line 473, in _get_service_account_credentials
    credentials = service_account.Credentials.from_service_account_info(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/google/oauth2/service_account.py", line 243, in from_service_account_info
    signer = _service_account_info.from_dict(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/google/auth/_service_account_info.py", line 57, in from_dict
    signer = crypt.RSASigner.from_service_account_info(data)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/google/auth/crypt/base.py", line 109, in from_service_account_info
    return cls.from_string(
           ^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/google/auth/crypt/_cryptography_rsa.py", line 133, in from_string
    private_key = serialization.load_pem_private_key(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3097, in signal_handler
    raise AirflowTaskTerminated("Task received SIGTERM signal")
airflow.exceptions.AirflowTaskTerminated: Task received SIGTERM signal
[2025-04-14T00:17:58.494+0000] {taskinstance.py:1226} INFO - Marking task as UP_FOR_RETRY. dag_id=weather_forecast_ingestion_dag, task_id=ingest, run_id=scheduled__2025-04-13T23:00:00+00:00, execution_date=20250413T230000, start_date=20250414T000001, end_date=20250414T001758
[2025-04-14T00:17:58.508+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-04-14T00:17:58.527+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=617, status='terminated', exitcode=2, started='00:00:00') (617) terminated with exit code 2
